src/v/kafka/server/errors.h

```diff
@@ -113,4 +114,48 @@ constexpr error_code map_topic_error_code(cluster::errc code) {
     return error_code::unknown_server_error;
 }
 
+constexpr error_code map_tx_errc(cluster::tx::errc ec) {
+    switch (ec) {
+    case cluster::tx::errc::none:
+        return error_code::none;
+    case cluster::tx::errc::leader_not_found:
+    case cluster::tx::errc::shard_not_found:
+    case cluster::tx::errc::partition_not_found:
+    case cluster::tx::errc::stm_not_found:
+    case cluster::tx::errc::partition_not_exists:
+    case cluster::tx::errc::not_coordinator:
+    case cluster::tx::errc::stale:
+        return error_code::not_coordinator;
+    case cluster::tx::errc::coordinator_not_available:
+        return error_code::coordinator_not_available;
+    case cluster::tx::errc::pid_not_found:
+        return error_code::unknown_producer_id;
+    case cluster::tx::errc::timeout:
+        return error_code::request_timed_out;
+    case cluster::tx::errc::conflict:
+        return error_code::invalid_txn_state;
+    case cluster::tx::errc::fenced:
+    case cluster::tx::errc::invalid_producer_epoch:
+        return error_code::invalid_producer_epoch;
+    case cluster::tx::errc::invalid_txn_state:
+        return error_code::invalid_txn_state;
+    case cluster::tx::errc::tx_id_not_found:
+    case cluster::tx::errc::tx_not_found:
+        return error_code::transactional_id_not_found;
+    case cluster::tx::errc::invalid_producer_id_mapping:
+        return error_code::invalid_producer_id_mapping;
+    case cluster::tx::errc::partition_disabled:
+        return error_code::replica_not_available;
+    case cluster::tx::errc::concurrent_transactions:
+        return error_code::concurrent_transactions;
+    case cluster::tx::errc::preparing_rebalance:
+    case cluster::tx::errc::rebalance_in_progress:
+    case cluster::tx::errc::coordinator_load_in_progress:
+    case cluster::tx::errc::request_rejected:
```

Comment:
these seem to be retriable at a high level but are mapped to USE which is terminal state and requires producer reset?